arrow-array/src/array/mod.rs

```diff
@@ -916,7 +916,7 @@ mod tests {
     #[test]
     fn test_null_struct() {
         let struct_type =
-            DataType::Struct(vec![Field::new("data", DataType::Int64, true)]);
+            DataType::Struct(vec![Field::new("data", DataType::Int64, false)]);
```

Comment:
This reverts a change in #3205 - you should be able to create a null struct containing a non-nullable child

Reply:
```suggestion
            // It is possible to create a null struct containing a non-nullable child 🤯 
            // see https://github.com/apache/arrow-rs/pull/3244 for details
            DataType::Struct(vec![Field::new("data", DataType::Int64, false)]);
```

It might be a good idea to leave some rationale for this very non obvious corner case